EKrut-Common/src/ekrut/net/ShipmentRequest.java

```diff
@@ -8,27 +8,27 @@
 public class ShipmentRequest implements Serializable {
 
 	private static final long serialVersionUID = 5704922451119200259L;
-	private OrderStatus action;
+	private OrderStatus status;
+	private ShipmentRequestType action;
 	private int orderId;
 	private String clientAddress;
 	private LocalDateTime date;
+	private String area;
 	
-	public ShipmentRequest(int orderId, String clientAddress, LocalDateTime date) {
-		this.action = OrderStatus.AWAITING_DELIVERY;
-		this.orderId = orderId;
-		this.clientAddress = clientAddress;
-		this.date = date;
-	}
 	
+	// Worker Confirm deliveries.
+	public ShipmentRequest(ShipmentRequestType action, String area) {
+		this.action = action;
+		this.area = area;
+	}
 	
 	public ShipmentRequest(OrderStatus awaitingDelivery, int orderId) {
-		this.action = awaitingDelivery;
+		this.status = awaitingDelivery;
 		this.orderId = orderId;
 	}
 	
-
-	public OrderStatus getAction() {
-		return action;
```

Comment:
You need this method because otherwise how are you going to find out what the request is requesting?

Reply:
I did not delete the method I have changed the name to "getStatus" but I need to add getAction as well yes, and I will fix the other comment